src/saveload/saveload.cpp

```diff
@@ -1368,7 +1364,6 @@ static inline size_t SlCalcDequeLen(const void *deque, VarType conv)
 		case SLE_VAR_U32: return SlStorageHelper<std::deque, uint32_t>::SlCalcLen(deque, conv);
 		case SLE_VAR_I64: return SlStorageHelper<std::deque, int64_t>::SlCalcLen(deque, conv);
 		case SLE_VAR_U64: return SlStorageHelper<std::deque, uint64_t>::SlCalcLen(deque, conv);
-		case SLE_VAR_STR: return SlStorageHelper<std::deque, std::string>::SlCalcLen(deque, conv, SL_STDSTR);
```

Comment:
Might be useful to retain the `case` with documentation that no  variable-sized elements may be in the `deque`, and similar to the variant in `SlCalcVectorLen`.
Maybe something along the lines of: `case SLE_VAR_STR: // Variable-sized elements require special handling for the save game to follow the specifications. Use a struct list instead.` should "scare" us from doing it again. And it would make it much easier to prevent such things in future reviews.